rados/rados_getaddrs.go

```diff
@@ -0,0 +1,31 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <stdlib.h>
+// #include <rados/librados.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// GetAddrs wraps the function to get the addresses of
+// the RADOS session, suitable for blocklisting.
+//
+// Implements:
+//
+//	int rados_getaddrs(rados_t cluster, char **addrs)
+func (c *Conn) GetAddrs() (string, error) {
```

Comment:
I was curious about the formatting of this string. I looked in the python and C header file in the ceph tree but nothing documents the format of this string. So I suppose it's meant to be opaque. That's OK.  If we ever needed to parse it in the future we can add a helper function for that later.

Reply:
I was also wondering about that, but if librados doesn't bother parsing the output into something manageable, then I guess we can treat it the same way?

* [`rados_getaddrs()`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/librados/librados_c.cc#L526-L535) calls `librados::RadosClient::get_addrs()`
* [`librados::RadosClient::get_addrs()`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/librados/RadosClient.cc#L777-L781) calls `*cos << messenger->get_myaddrs();`
  * [`get_myaddrs()`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/msg/Messenger.h#L221-L229) returns [`entity_addrvec_t`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/msg/msg_types.h#L581-L750)
    * [`friend std::ostream& operator<<(std::ostream& out, const entity_addrvec_t& av)`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/msg/msg_types.h#L728-L736) -- here, if the vector has only a single item it prints only that one entity address:
      * Using this function [`std::ostream& operator<<(std::ostream& out, const entity_addr_t &addr)`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/msg/msg_types.cc#L220-L230)
    * If the vector has more items, it uses formats it into `[item1,item2,...]` using [`inline std::ostream& operator<<(std::ostream& out, const std::vector<A,Alloc>& v)`](https://github.com/ceph/ceph/blob/66bf56ec50d0e1688ffc2601a12a365d94466b90/src/include/types.h#L134-L145)

Let me know if you'd like me also to parse the output in some way.

Reply:
So far I got only a single entity address, so there is no weird formatting and only the `<IP>:<port>/nonce` is returned, but I guess there can be more.

Reply:
OK, let's assume the content is informational and/or opaque.